PINCache/PINDiskCache.m

```diff
@@ -72,41 +67,24 @@ - (instancetype)initWithName:(NSString *)name
     return [self initWithName:name rootPath:[NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES) objectAtIndex:0]];
 }
 
-- (instancetype)initWithName:(NSString *)name rootPath:(NSString *)rootPath
-{
-    if (!name)
+- (instancetype)initWithName:(NSString *)name rootPath:(NSString *)rootPath {
+    if (!name) {
         return nil;
-    
+    }
     if (self = [super init]) {
-        _name = [name copy];
-        _asyncQueue = dispatch_queue_create([[NSString stringWithFormat:@"%@ Asynchronous Queue", PINDiskCachePrefix] UTF8String], DISPATCH_QUEUE_CONCURRENT);
-        _lockSemaphore = dispatch_semaphore_create(1);
-        _willAddObjectBlock = nil;
-        _willRemoveObjectBlock = nil;
-        _willRemoveAllObjectsBlock = nil;
-        _didAddObjectBlock = nil;
-        _didRemoveObjectBlock = nil;
-        _didRemoveAllObjectsBlock = nil;
-        
-        _byteCount = 0;
-        _byteLimit = 0;
-        _ageLimit = 0.0;
-        
-        _dates = [[NSMutableDictionary alloc] init];
-        _sizes = [[NSMutableDictionary alloc] init];
-        
-        NSString *pathComponent = [[NSString alloc] initWithFormat:@"%@.%@", PINDiskCachePrefix, _name];
-        _cacheURL = [NSURL fileURLWithPathComponents:@[ rootPath, pathComponent ]];
-        
-        //we don't want to do anything without setting up the disk cache, but we also don't want to block init, it can take a while to initialize
-        //this is only safe because we use a dispatch_semaphore as a lock. If we switch to an NSLock or posix locks, this will *no longer be safe*.
-        [self lock];
-        dispatch_async(_asyncQueue, ^{
-            [self createCacheDirectory];
-            [self initializeDiskProperties];
+        _backgroundTaskClass = [PINApplicationBackgroundTask class];
```

Comment:
Crazy idea: we could do something this at runtime:
```
if ([[[NSBundle mainBundle] bundlePath] hasSuffix:@".appex"]) {
    _backgroundTaskClass = nil;
} else {
    _backgroundTaskClass = NSClassFromString(@"PINApplicationBackgroundTask");
}
```

I know it's crazy and throws away some of the compile time guarantees, but jeeze Apple is it too much to ask for some way to tailor code for extensions at runtime?